examples/morpheusvm/actions/transfer_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package actions
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/examples/morpheusvm/storage"
+	"github.com/ava-labs/hypersdk/state"
+	"github.com/ava-labs/hypersdk/tstate"
+)
+
+func TestTransferAction(t *testing.T) {
+	require := require.New(t)
+	ts := tstate.New(1)
+	emptyBalanceKey := storage.BalanceKey(codec.EmptyAddress)
+
+	tests := map[string]chain.ActionTest{
+		"ZeroTransfer": {
+			Action: &Transfer{
+				To:    codec.EmptyAddress,
+				Value: 0,
+			},
+			ExpectedErr: ErrOutputValueZero,
+		},
+		"InvalidStateKey": {
+			Action: &Transfer{
+				To:    codec.EmptyAddress,
+				Value: 1,
+			},
+			State:       ts.NewView(map[string]state.Permissions{}, map[string][]byte{}),
+			ExpectedErr: tstate.ErrInvalidKeyOrPermission,
+		},
+		"NotEnoughBalance": {
+			Action: &Transfer{
+				To:    codec.EmptyAddress,
+				Value: 1,
+			},
+			State: func() state.Mutable {
+				keys := make(state.Keys)
+				keys.Add(string(emptyBalanceKey), state.Read)
+				tsv := ts.NewView(keys, map[string][]byte{})
+				return tsv
+			}(),
+			ExpectedErr: storage.ErrInvalidBalance,
+		},
+		"SimpleTransfer": {
+			Action: &Transfer{
+				To:    codec.EmptyAddress,
+				Value: 1,
+			},
+			State: func() state.Mutable {
+				keys := make(state.Keys)
+				store := chain.NewInMemoryStore()
+				require.NoError(storage.SetBalance(context.TODO(), store, codec.EmptyAddress, 1))
```

Comment:
`context.Background()` is the [correct value](https://google.github.io/styleguide/go/decisions#keep-going:~:text=In%20tests%3A%20TestXXX%2C%20BenchmarkXXX%2C%20FuzzXXX) to use here.

`TODO()` is generally used for when a `Context` isn't available and a future refactoring is needed to introduce it. I think it's so you can do a global plain-text search for the phrase `TODO`.

Reply:
Thanks. Replaced the `context.TODO()` by a `context.Background()`.